File: ts/src/push-to-var/fs/cfg-adapt/bar-bf-cfg-adapter.ts
/*
 * Copyright © 2022 Paulo Villela. All rights reserved.
 * Use of this source code is governed by the MIT license
 * that can be found in the LICENSE file.
 */

import { AppCfgInfo } from "../../config/app-cfg";
import { barBfCfgSrc, BarBfCfgInfo } from "../bar-bf";
import { CfgSrcAdaptation } from "../../fwk/cfg-src";

function adapter(appCfg: AppCfgInfo): BarBfCfgInfo {
  return {
    z: appCfg.y
  };
}

export const barBfCfgAdaptation = new CfgSrcAdaptation(
  barBfCfgSrc,
  adapter,
)

File: rs/hold/pushtovar/fs/cfgadapt/BarBfCfgAdapter.kt
package pushtovar.fs.cfgadapt

import pushtovar.config.AppCfgInfo
import pushtovar.fs.BarBfCfgInfo
import pushtovar.fs.barBfCfgSrc
import pushtovar.fwk.CfgSrcAdaptation

fun barBfCfgAdapter(appCfg: AppCfgInfo): BarBfCfgInfo {
	return BarBfCfgInfo(
        z = appCfg.y,
    )
}

val barBfCfgAdaptation = CfgSrcAdaptation(
	barBfCfgSrc,
	::barBfCfgAdapter,
)
